### PR TITLE
[jdbc] Add warning about using tableCaseSensitiveItemNames without dedicated schema

### PR DESCRIPTION
**Pull request by jlaur**
Follow-up on #13544 and #9339 ([this comment](https://github.com/openhab/openhab-addons/issues/9339#issuecomment-1314923225)).

### CODE REVIEW

File: bundles/org.openhab.persistence.jdbc/README.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ This service can be configured in the file `services/jdbc.cfg`.
...
-| tableCaseSensitiveItemNames | `false`                                                      |    No     | table name case when `tableUseRealItemNames` is `true`. When set to `true`, item name case is preserved in table names and no suffix is used. When set to `false`, table names are lower cased and a numeric suffix is added. Please read [this](#case-sensitive-item-names) before enabling. |
+| tableCaseSensitiveItemNames | `false`                                                      |    No     | table name case when `tableUseRealItemNames` is `true`. When set to `true`, item name case is preserved in table names and no prefix or suffix is added. Please make sure to have a dedicated schema when using this option, since otherwise table name collisions are more likely to happen. When set to `false`, table names are lower cased and a numeric suffix is added. Please read [this](#case-sensitive-item-names) before enabling. |
```
Comment (lsiepel) [inline bundles/org.openhab.persistence.jdbc/README.md:65]: "table name case when `tableUseRealItemNames` is **`true`."**

I find this hard to understand, isn't it better to rephrase it to:
"This setting is only applicable when `tableUseRealItemNames` is **`true`."*"

Or does it mean soimething different?

Comment (jlaur) [reply bundles/org.openhab.persistence.jdbc/README.md]: Correct, thanks - rephrased.

**Review by lsiepel — APPROVED** [review]

**Review by lolodomo — APPROVED** [review]
LGTM
